File: FE/WEB/app/models/ticket.js
import Model, { attr, belongsTo, hasMany } from '@ember-data/model';

export default class TicketModel extends Model {
  @belongsTo('user', { async: true, inverse: null }) user;
  @belongsTo('tenant', { async: true, inverse: null }) tenant;
  @belongsTo('tenant', { async: true, inverse: null }) tenantDestination;
  //@belongsTo('ticketTag', { async: true, inverse: null }) ticketTag;
  @hasMany('ticket-tag-mapping', { async: true, inverse: 'ticket' })
  ticketTagMappings;
  @hasMany('ticket-message', { async: true, inverse: 'ticket' }) ticketMessages;
  @hasMany('ticket-to-do', { async: true, inverse: 'ticket' }) ticketToDos;
  @hasMany('ticket-operator-grant', { async: true, inverse: 'ticket' })
  ticketOperatorGrants;
  @hasMany('ticket-relation', { async: true, inverse: 'fatherTicket' })
  ticketRelationsAsFather;
  @hasMany('ticket-relation', { async: true, inverse: 'childTicket' })
  ticketRelationsAsChild;
  @hasMany('ticket-to-do', { async: true, inverse: 'ticket' }) ticketToDos;
  @hasMany('ticket-project-mapping', { async: true, inverse: 'ticket' })
  ticketProjectMappings;

  @attr('string', { defaultValue: '' }) email;
  @attr('string', { defaultValue: '' }) phone;
  @attr('string', { defaultValue: '' }) organization;
  @attr('string', { defaultValue: '' }) vat;

  @attr('string', { defaultValue: '0' }) assignedId; // id dell'utente operatore che gestisce la richiesta
  @attr('string', { defaultValue: '' }) userId; // id dell'utente che ha creato la richiesta
  @attr('number', { defaultValue: 0 }) tenantId; // tenant dell'azienda con cui è stata creata la richiesta
  @attr('number', { defaultValue: 0 }) tenantDestinationId; // tenant dell'azienda a cui è destinata la richiesta di assistenza

  @attr('string', { defaultValue: '' }) message;
  @attr('string', { defaultValue: 'unprocessed' }) status;

  @attr('number', { defaultValue: 0 }) priority; // importanza della richiesta, per chi la gestisce. 0 = non specificata, 3 = massima.
  @attr('number', { defaultValue: 0 }) priorityOwner; // importanza della richiesta, per chi l'ha creata. 0 = non specificata, 3 = massima.
  @attr('string', { defaultValue: '' }) closedById; // id dell'utente che ha impostato il ticket come completato
  @attr('date') closedDate; // data di chiusura
  @attr('string', { defaultValue: '' }) note;
  @attr('string', { defaultValue: '' }) answer;

  @attr('string', { defaultValue: '' }) organizationToBeConfirmed; // ragione sociale inserita dagli utenti anonimi
  @attr('string', { defaultValue: '' }) firstName;
  @attr('string', { defaultValue: '' }) lastName;

  @attr('date', {
    defaultValue() {
      return new Date();
    },
  })
  creationDate;

  @attr('number', { defaultValue: 0 }) number;
  //@attr('string', { defaultValue: '' }) ticketTagId;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/Ticket.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using MIT.Fwk.Core.Models;
using System;
using MIT.Fwk.Core.Attributes;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    [SkipJwtAuthentication(JwtHttpMethod.GET)]
    public class Ticket : Identifiable<string>
    {
        [Attr]
        public string Email { get; set; }
        [Attr]
        public string Organization { get; set; }
        [Attr]
        public string Vat { get; set; }
        [Attr]
        public string Status { get; set; }
        [Attr]
        public string Message { get; set; }
        [Attr]
        public DateTime CreationDate { get; set; }
        [Attr]
        public string AssignedId { get; set; }
        [Attr]
        public string Phone { get; set; }
        [Attr]
        public string UserId { get; set; }
        [Attr]
        public int TenantId { get; set; }
        [Attr]
        public int Priority { get; set; }
        [Attr]
        public string ClosedById { get; set; }
        [Attr]
        public string Note { get; set; }
        [Attr]
        public DateTime? ClosedDate { get; set; }
        [Attr]
        public string Answer { get; set; }
        [Attr]
        public string FirstName { get; set; }
        [Attr]
        public string LastName { get; set; }
        [Attr]
        public string OrganizationToBeConfirmed { get; set; }
        [Attr]
        [IdentityDB]
        public int? Number { get; set; }
        [Attr]
        public string TicketTagId { get; set; }


        [HasOne]
        public virtual User User { get; set; }
        [HasOne]
        public virtual User Assigned { get; set; }
        [HasOne]
        public virtual Tenant Tenant { get; set; }
        [HasOne]
        public virtual TicketTag TicketTag { get; set; }





    }
}
